Iterable-iOS-SDK/IterableAPI.m

```diff
@@ -463,21 +459,36 @@ - (void)trackInAppOpen:(NSNumber*)campaignId templateId:(NSNumber*)templateId me
 }
 
 // documented in IterableAPI.h
-- (void)trackInAppClick:(NSNumber*)campaignId templateId:(NSNumber*)templateId messageId:(NSString *)messageId buttonIndex:(NSNumber*)buttonIndex {
+- (void)inAppConsume:(NSString *)messageId {
+    NSDictionary *args;
+    
+    if (_email != nil) {
+        args = @{
+                 ITBL_KEY_EMAIL: self.email,
+                 ITBL_KEY_MESSAGE_ID: messageId
+                 };
+    } else {
+        args = @{
+                 ITBL_KEY_USER_ID: self.userId,
+                 ITBL_KEY_MESSAGE_ID: messageId
+                 };
+    }
+    NSURLRequest *request = [self createRequestForAction:ENDPOINT_INAPP_CONSUME withArgs:args];
+    [self sendRequest:request onSuccess:[IterableAPI defaultOnSuccess:@"inAppConsume"] onFailure:[IterableAPI defaultOnFailure:@"inAppConsume"]];
+}
+
+// documented in IterableAPI.h
+- (void)trackInAppClick:(NSString *)messageId buttonIndex:(NSNumber*)buttonIndex {
     NSDictionary *args;
     if (_email != nil) {
         args = @{
                  ITBL_KEY_EMAIL: self.email,
-                 ITBL_KEY_CAMPAIGN_ID: campaignId,
```

Comment:
Sounds like I might have to modify this logic since we remove the redis mapping after we make a call to consume. So any inAppClicks made after InAppConsume would fail.